src/toast/ops/polyfilter.py

```diff
@@ -269,13 +269,10 @@ def _exec(self, data, detectors=None, **kwargs):
             signal_mem = AlignedF64()
             coeff_mem = AlignedF64()
 
-            views = temp_ob.view[self.view]
+            views = temp_ob.intervals[self.view]
             for iview, view in enumerate(views):
-                if view.start is None:
-                    # This is a view of the whole obs
-                    nsample = temp_ob.n_local_samples
-                else:
-                    nsample = view.stop - view.start
+                nsample = view.last - view.first + 1
+                vslice = slice(view.first, view.last + 1)
```

Comment:
That is a fair point.  Originally, the "view" concept (i.e. returning a list of detdata / shared views) seemed like a nice shorthand way of accessing the underlying data- especially when working interactively.  However, depending on the underlying data, there can be a performance penalty looping over views first and then detectors within a view (since this is jumping around in memory).  The operators where we have ported the internals to C++ / OpenMP or Jax now always loop over detectors and then elements of the intervals list (which is just a numpy array with a custom dtype).  This keeps the memory access contiguous.

I know there are many operators that need updated to this model, but I was going to avoid doing that in this PR (which has enough already!).  I like your suggestion here.  So the intervals object is the plain numpy array of intervals, and the view object would not only provide the current list of detdata / shared views (which it implements via `Sequence.__getitem__`), but also have a method that returns a list of actual `slice` objects.

So the pattern would look something like:
```
for det in dets:
    for vslice in ob.view["intervals_name"].slices:
        nsample = vslice.stop - vslice.start
        ob.detdata["signal"][det, vslice] += 1.0
```
Instead of
```
for det in dets:
    for intrvl in ob.intervals["intervals_name"]:
        vslice = slice(intrvl.first, intrvl.last + 1)
        nsample = intrvl.last - intrvl.first + 1
        ob.detdata["signal"][det, vslice] += 1.0
```

Reply:
I am going to open a separate issue to track this and mark this conversation as resolved.